packages/yumemi_lints/lib/src/command_services/update_command_service.dart

```diff
@@ -0,0 +1,169 @@
+        case ProjectType.dart:
+          version = getDartVersion(_getPubspecFile());
+          break;
+        case ProjectType.flutter:
+          version = getFlutterVersion(_getPubspecFile());
+          break;
+      }
+
+      final includeLine =
+          'include: package:yumemi_lints/${projectType.name}/${version.excludePatchVersion}/recommended.yaml';
+      _updateAnalysisOptionsFile(includeLine);
+      return ExitStatus.success;
+    } on FormatException catch (e) {
+      print(e.message);
+      return ExitStatus.error;
+    }
+  }
+
+  File _getPubspecFile() {
+    final pubspecFile = File(
+      path.join(
+        Directory.current.path,
+        'pubspec.yaml',
+      ),
+    );
+
+    if (!pubspecFile.existsSync()) {
+      throw Exception(
+        'The pubspec.yaml file could not be found in the current directory. '
+        'Please ensure that you are in the correct project directory '
+        'or create a pubspec.yaml file if it does not exist.',
+      );
+    }
+
+    return pubspecFile;
+  }
+
+  ProjectType _getProjectType() {
+    final pubspecFile = _getPubspecFile();
+    final lines = pubspecFile.readAsLinesSync();
+    for (final line in lines) {
+      if (line.trim() == 'sdk: flutter') {
+        return ProjectType.flutter;
+      }
+    }
+    return ProjectType.dart;
+  }
+
+  @visibleForTesting
+  Version getFlutterVersion(File pubspecFile) {
+    final yaml = loadYaml(pubspecFile.readAsStringSync()) as YamlMap;
+    final flutterVersion = (yaml['environment'] as YamlMap)['flutter'];
+
+    if (flutterVersion == null) {
+      throw FormatException(
+        'Please put [flutter: Flutter Version] '
+        'under the environment section in pubspec.yaml',
+      );
+    }
+
+    return extractVersion(flutterVersion as String);
+  }
+
+  @visibleForTesting
+  Version getDartVersion(File pubspecFile) {
+    final yaml = loadYaml(pubspecFile.readAsStringSync()) as YamlMap;
+    final dartVersion = (yaml['environment'] as YamlMap)['sdk'];
+
+    if (dartVersion == null) {
+      throw FormatException(
+        'Please put [sdk: Dart Version] '
+        'under the environment section in pubspec.yaml',
+      );
+    }
+
+    return extractVersion(dartVersion as String);
+  }
+
+  @visibleForTesting
+  Version extractVersion(String versionString) {
+    if (versionString.contains('<') && !versionString.contains('>=')) {
+      throw FormatException('Please specify the minimum version.');
+    }
+
+    RegExp regExp;
+    if (versionString.contains('>=')) {
+      regExp = RegExp(r'>=\s*(\d+\.\d+\.\d+)');
+    } else {
+      regExp = RegExp(r's*(\d+\.\d+\.\d+)');
+    }
+    final match = regExp.firstMatch(versionString);
+
+    if (match == null) {
+      throw FormatException(
+        'Version could not be found from pubspec.yaml.',
+      );
+    }
+
+    final version = match.group(1);
+    if (version == null) {
+      throw const FormatException('Version extraction failed.');
+    }
```

Comment:
![nits-badge](https://img.shields.io/badge/review-nits-orange.svg) 
Here, it should not be possible for the version to become null, so it would be better to do the following as it would increase readability.
```suggestion
    final version = match.group(1)!;
```

Reply:
Thanks for pointing that out.
I fixed. 🙏 177335b4819ae1a33a8aa4982cc6ba902ed903ce